SlaveWAMPServer.spec.js

```diff
@@ -161,60 +167,247 @@ describe('SlaveWAMPServer', () => {
+			slaveWAMPServer.sendToMaster(validProcedure, validData, validSocketId, validCb);
+			expect(slaveWAMPServer.interProcessRPC).to.have.nested.property(`${validSocketId}.${validProcedure}.${validSignature}`).to.be.a('function');
+		});
+	});
+
+	describe('onSocketDisconnect', () => {
+		it('should leave interProcessRPC in initial state when invoked without arguments', () => {
+			slaveWAMPServer.onSocketDisconnect();
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should leave interProcessRPC in initial state when entry does not exist', () => {
+			slaveWAMPServer.onSocketDisconnect(validSocketId);
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should remove existing entry from interProcessRPC', () => {
+			slaveWAMPServer.interProcessRPC = validInterProcessRPCEntry;
+			slaveWAMPServer.onSocketDisconnect(validSocketId);
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+	});
+
+	describe('saveCall', () => {
+		it('should throw an error when invoked without arguments', () => {
+			expect(slaveWAMPServer.saveCall).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without socket id', () => {
+			delete validRequest.socketId;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without procedure', () => {
+			delete validRequest.procedure;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without signature', () => {
+			delete validRequest.signature;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without callback', () => {
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest);
+			}).to.throw('Cannot save a call without callback');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should create a new entry in interProcessRPC for valid request', () => {
+			slaveWAMPServer.saveCall(validRequest, validCb);
+			expect(slaveWAMPServer.interProcessRPC).to.have.nested.property(`${validSocketId}.${validProcedure}.${validSignature}`).to.be.a('function');
+		});
+
+		it('should create multiple entries in interProcessRPC for multiple valid requests', () => {
+			slaveWAMPServer.saveCall(validRequest, validCb);
+			const validRequestB = validRequest;
```

Comment:
If you're happy to mutate the object, just use `validRequest`. If you want `validRequestB` to be a different object, create a new one.